component/loki/process/stages/extensions.go

```diff
@@ -24,7 +27,37 @@ type DockerConfig struct{}
 
 // CRIConfig is an empty struct that is used to enable a pre-defined pipeline
 // for decoding entries that are using the CRI logging format.
-type CRIConfig struct{}
+type CRIConfig struct {
+	MaxPartialLines            int              `river:"max_partial_lines,attr,optional"`
+	MaxPartialLineSize         flagext.ByteSize `river:"max_partial_line_size,attr,optional"`
+	MaxPartialLineSizeTruncate bool             `river:"max_partial_line_size_truncate,attr,optional"`
+}
+
+var (
+	_ river.Defaulter = (*CRIConfig)(nil)
+	_ river.Validator = (*CRIConfig)(nil)
+)
+
+// DefaultCRIConfig applies the default values on
+var DefaultCRIConfig = CRIConfig{
+	MaxPartialLines:            100,
+	MaxPartialLineSize:         0,
+	MaxPartialLineSizeTruncate: false,
```

Comment:
Ah, I was under impression that you can set `max_partial_line_size_truncate` to `false` and then the lines that are longer than `max_partial_line_size` will be just dropped instead of truncated. That was the only logical reason to have two config options, but that's not how this works and indeed they are redundant as you've mentioned in the comment above.